_posts/2022-07-06-how-to-use-airflow-to-schedule-trino-jobs.md

```diff
@@ -0,0 +1,376 @@
+Once you have done that you need to restart all five containers:
+
+```
+$ docker container restart <container_id_1> ... <container_id_5>
+```
+
+# Creating a Trino connection
+
+After you have installed the TrinoHook and restarted Airflow you can create a
+connection to your Trino cluster through the Airflow web UI.  If you just
+install Airflow then go to `http://localhost:8080` on your browser and login.
+The default credentials unless changed are `airflow` for username and password.
+
+Go to Admin > Connections.
+
+<p align="center">
+   <img align="center" width="75%" src="/assets/blog/trino-airflow-blog/airflow-connections.png"/>
+</p>
+
+Click on the blue button to **Add a new record**.
+
+<p align="center">
+   <img align="center" width="75%" src="/assets/blog/trino-airflow-blog/airflow-new-connection.png"/>
+</p>
+
+Select **Trino** from the Connection Type dropdown and provide the following information:
+
+<table>
+  <tr>
+   <td>Connection Id</td>
+   <td>Whatever you want to call your connection.</td>
+  </tr>
+  <tr>
+   <td>
+    Host
+   </td>
+   <td>The hostname or host ip of your trino cluster, e.g., <code>localhost</code>, <code>10.10.10.1</code>, or <code>www.mytrino.com</code></td>
+  </tr>
+  <tr>
+   <td>Schema</td>
+   <td>A schema in your Trino cluster, not important which.</td>
+  </tr>
+  <tr>
+   <td>Login</td>
+   <td>The username of the user that Airflow will use to connect to Trino.  Best practice would be to create a service account like ‘airflow’. Just understand that this user access level ise used to execute SQL statements in Trino.</td>
+  </tr>
+  <tr>
+   <td>Password</td>
+   <td>The password of the user that Airflow will use to connect to Trino if authentication is enabled.</td>
+  </tr>
+  <tr>
+   <td>Port</td>
+   <td>The port where the Trino Web UI can be accessed, e.g., <code>8080</code>, <code>8443</code>.</td>
+  </tr>
+  <tr>
+   <td>Extra</td>
+   <td>Additional settings, like <code>protocol:https</code> if using TLS, or <code>verify:false</code> if you are using a self-signed certificate.</td>
+  </tr>
+</table>
+
+Be aware that the test button might not actually return any feedback for Trino connections.
+
+<p align="center">
+   <img align="center" width="50%" src="/assets/blog/trino-airflow-blog/airflow-add-connection.png"/>
+</p>
+
+# Deploying a TrinoOperator
+
+At the time of writing this article there is no TrinoOperator, so you have to
+write your own.  You find an implementation in the following section, to get you started.  This operator allows you to
+execute any SQL statements that Trino supports such as `SELECT`, `INSERT`, `CREATE`, `SET SESSION`, and others. You can run multiple statements in a single task so
+they are part of a single Trino session.
+
+To create the TrinoOperator use your favorite text editor to create a file called
+`trino_operator.py` with the following code in it and place it in the
+`airflow/plugins` directory you created earlier.  As you can see it is written
```

Comment:
drop the sentence "As you can see it is written .." since this is in the next paragraph